// vant/lib/collapse/index.js
'use strict';

exports.__esModule = true;

var _create = require('../utils/create');

var _create2 = _interopRequireDefault(_create);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = (0, _create2.default)({
  render: function render() {
    var _vm = this;var _h = _vm.$createElement;var _c = _vm._self._c || _h;return _c('div', { staticClass: "van-collapse van-hairline--top-bottom" }, [_vm._t("default")], 2);
  },

  name: 'collapse',

  model: {
    prop: 'activeNames'
  },

  props: {
    accordion: Boolean,
    activeNames: [String, Number, Array]
  },

  data: function data() {
    return {
      items: []
    };
  },


  methods: {
    switch: function _switch(name, expanded) {
      var activeNames = this.activeNames;

      if (!this.accordion) {
        name = expanded ? activeNames.concat(name) : activeNames.filter(function (activeName) {
          return activeName !== name;
        });
      }
      this.$emit('change', name);
      this.$emit('input', name);
    }
  }
});
